local-cloudbuild.py

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env python
+#
+# Run the steps of cloudbuild.yaml locally, emulating Google Container
+# Builder functionality.  Does not actually push images to the
+# Container Registry.  Not all functionality is supported.
+#
+# Based on https://cloud.google.com/container-builder/docs/api/build-steps
+
+# System packages
+import argparse
+import getpass
+import os
+import shutil
+import subprocess
+import sys
+import tempfile
+
+# Third party packages
+import yaml
+
+
+# Types
+class CloudBuildError(Exception):
```

Comment:
Seems like `ValueError` can be used instead of this pretty much everywhere.

(In general, I avoid defining custom exception classes unless really necessary)